### DISCUSSION TITLE
behavior if read_terragrunt_config(find_in_parent_folders("xxxx.hcl")) found multiple values

### DISCUSSION THREAD

**Discussion by kholisrag** [gruntwork-io/knowledge-base#528] [Help] [answered]
whats the behavior for these function

`terragrunt.hcl`
```hcl
locals {
  service_vars = read_terragrunt_config(find_in_parent_folders("service.hcl"))
}
```

`service.hcl`
```hcl
locals {
  service = basename(get_path_from_repo_root())
}

```

if we have multiple service.hcl, with same key but different value?

lets say we have directory structure like : 

```
vault
├── service.hcl <-- return `vault` as value
└── vault-client
    ├── asg
    |   └── terragrunt.hcl
    └─service.hcl <-- return `vault-client` as value
```

if its like that whats the `local.service_vars.locals.service` actual value?

can anyone enlighten me related to this case? 

---

<ins datetime="2022-08-05T11:04:55Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109103">Tracked in ticket #109103</a></p>
</ins>

**Comment by pete0emerson** [comment, marked as answer]
This is documented in our [Built-in functions](https://terragrunt.gruntwork.io/docs/reference/built-in-functions/#find_in_parent_folders) secion of our Terragrunt documentation.

**Reply by kholisrag** [reply]
Still not quite understand after reading n working with terragrunt quite few days already tho,

Just want to confirm after trying seem the value will be vault-server

**Reply by pete0emerson** [reply]
@kholisrag what have you tried? Did you run `plan` to see what the results would be? Are you seeing behavior that is different than the documentation?

```
$ tree modules vault --noreport
modules
└── sample
    ├── main.tf
    ├── outputs.tf
    └── variables.tf
vault
├── service.hcl
└── vault-client
    ├── asg
    │   └── terragrunt.hcl
    └── service.hcl

$ rg '' modules
modules/sample/variables.tf
1:variable "service" {
2:	type = string
3:}

modules/sample/main.tf
1:resource "null_resource" "main" {}

modules/sample/outputs.tf
1:output "service" {
2:	value = var.service
3:}


$ rg '' vault
vault/service.hcl
1:locals {
2:  service = basename(get_path_from_repo_root())
3:}

vault/vault-client/asg/terragrunt.hcl
1:terraform {
2:   source = ""
3:}
4:
5:locals {
6:  service_vars = read_terragrunt_config(find_in_parent_folders("service.hcl"))
7:}
8:
9:inputs = {
10:	service = local.service_vars.locals.service
11:}

vault/vault-client/service.hcl
1:locals {
2:  service = basename(get_path_from_repo_root())
3:}

$ cd vault/vault-client/asg

$ terragrunt init --terragrunt-source ../../../modules//sample

$ terragrunt plan --terragrunt-source ../../../modules//sample | grep -A1 'Changes to Outputs'
Changes to Outputs:
  + service = "vault-client"

$ rm ../service.hcl

$ terragrunt plan --terragrunt-source ../../../modules//sample | grep -A1 'Changes to Outputs'
Changes to Outputs:
  + service = "vault"
```

**Reply by kholisrag** [reply]
Seem we're in the same result just figure out yesterday, and its like what you did,
Actually the error I encounter before isn't because of this stupid question / problem, its another. But nwm tho.

Thank you for helping n confirming it, I truly appreciate it.
